Mezzo GA (#10047)
microsoftgraph/microsoft-graph-docs: api-reference/v1.0/api/group-list-transitivemembers.md

## Optional query parameters

> This method supports the [OData query parameters](/graph/query-parameters) to help customize the response, including `$search`, `$count`, and `$filter`. You can use `$search` on the **displayName** property. When items are added or updated for this resource, they are specially indexed for use with the `$count` and `$search` query parameters. There can be a slight delay between when an item is added or updated and when it is available in the index.

Reviewer: ```suggestion
This method supports the [OData query parameters](/graph/query-parameters) to help customize the response, including `$search`, `$count`, and `$filter`. You can use `$search` on the **displayName** and **description** properties. When items are added or updated for this resource, they are specially indexed for use with the `$count` and `$search` query parameters. There can be a slight delay between when an item is added or updated and when it is available in the index.
```

{
  "value": [
    {

>       "id": "11111111-2222-3333-4444-555555555555",

Reviewer: ```suggestion
      "@odata.type": "#microsoft.graph.user",
      "id": "11111111-2222-3333-4444-555555555555",
```

>       "mail": "group1@contoso.com",

Reviewer: ```suggestion
     "displayName": "Joseph Price",
     "mail": "Joseph.Price@contoso.com",
```

      "mailEnabled": true,

>       "mailNickname": "Contoso1",

Reviewer: remove mailEnabled, mailNickname and securityEnabled: not relevant for User Object

      "securityEnabled": true
    }
  ]
}
```

### Example 2: Get only a count of transitive membership

#### Request

The following is an example of the request.

<!-- {
  "blockType": "request",
  "name": "get_group_transitivemembers_count"
}-->

```msgraph-interactive
GET https://graph.microsoft.com/v1.0/groups/{id}/transitiveMembers/$count
ConsistencyLevel: eventual
```

#### Response

The following is an example of the response.

>**Note:** The response object shown here might be shortened for readability. All the properties will be returned from an actual call.

<!-- {
  "blockType": "response",
  "truncated": true,

>   "@odata.type": "microsoft.graph.directoryObject",

Reviewer: raw count not a directory object

  "isCollection": true
} -->
```http
HTTP/1.1 200 OK
Content-type: text/plain
```

> 893

Reviewer: ```suggestion
`893`
```

### Example 3: Use OData cast and $search to get membership in groups with display names that contain the letters 'tier' including a count of returned objects

#### Request

The following is an example of the request.

<!-- {
  "blockType": "request",
  "name": "get_tier_count"
}-->
```msgraph-interactive
GET https://graph.microsoft.com/v1.0/groups/{id}/transitiveMembers/microsoft.graph.user?$count=true&$orderBy=displayName&$search="displayName:tier"&$select=displayName,id
ConsistencyLevel: eventual
```

#### Response

The following is an example of the response.

>**Note:** The response object shown here might be shortened for readability. All the properties will be returned from an actual call.

<!-- {
  "blockType": "response",
  "truncated": true,

>   "@odata.type": "microsoft.graph.group",

Reviewer: ```suggestion
  "@odata.type": "microsoft.graph.user",
```

  "isCollection": true
} -->
```http
HTTP/1.1 200 OK
Content-type: application/json

{

>   "@odata.context":"https://graph.microsoft.com/v1.0/$metadata#groups(displayName,id)",

Reviewer: ```suggestion
  "@odata.context":"https://graph.microsoft.com/v1.0/$metadata#users(displayName,id)",
```

  "@odata.count":7,
  "value":[
    {

>       "displayName":"Contoso-tier Query Notification",

Reviewer: ```suggestion
      "displayName":"Joseph Price",
```

      "id":"11111111-2222-3333-4444-555555555555"
    }
  ]
}
```

### Example 4: Use OData cast and $filter to get user membership in groups with a display name that starts with 'A' including a count of returned objects

#### Request

The following is an example of the request.

<!-- {
  "blockType": "request",
  "name": "get_a_count"
}-->

```msgraph-interactive
GET https://graph.microsoft.com/v1.0/groups/{id}/transitiveMembers/microsoft.graph.users?$count=true&$orderBy=displayName&$filter=startswith(displayName, 'a')
ConsistencyLevel: eventual
```

#### Response

The following is an example of the response.

>**Note:** The response object shown here might be shortened for readability. All the properties will be returned from an actual call.

<!-- {
  "blockType": "response",
  "truncated": true,

>   "@odata.type": "microsoft.graph.group",

Reviewer: ```suggestion
  "@odata.type": "microsoft.graph.user",
```

  "isCollection": true
} -->
```http
HTTP/1.1 200 OK
Content-type: application/json

{

>   "@odata.context":"https://graph.microsoft.com/v1.0/$metadata#groups",

Reviewer: ```suggestion
  "@odata.context":"https://graph.microsoft.com/v1.0/$metadata#users",
```

  "@odata.count":76,
  "value":[
    {

>       "displayName":"AAD Contoso Users",

Reviewer: ```suggestion
      "displayName":"Joseph Price",
```

>       "mail":"AADContoso_Users@contoso.com",

Reviewer: ```suggestion
      "mail":"Joseph.Price@contoso.com",
```

      "mailEnabled":true,

>       "mailNickname":"AADContoso_Users",

Reviewer: remove mailEnabled, mailNickname and securityEnabled: not relevant for User Object
